### PR TITLE
Add mAP to tensorboard when using train.py

### PR DESCRIPTION
**Pull request by mxvs**
If the user enables Tensorboard in `train.py` with `--tensorboard-dir` (introduced with #267) and has not disabled evaluation with `--no-evaluation` the mean_ap value will now be automatically added to TensorBoard (as introduced in #278).

### CODE REVIEW

File: keras_retinanet/bin/train.py [hunk 1]
```diff
@@ -109,14 +109,31 @@ def create_callbacks(model, training_model, prediction_model, validation_generat
         checkpoint = RedirectModel(checkpoint, prediction_model)
         callbacks.append(checkpoint)
 
+    if args.tensorboard_dir:
+        tensorb = keras.callbacks.TensorBoard(
```
Comment (hgaiser) [inline keras_retinanet/bin/train.py]: I prefer the name `tensorboard` instead.

File: keras_retinanet/bin/train.py [hunk 2]
```diff
@@ -109,14 +109,31 @@ def create_callbacks(model, training_model, prediction_model, validation_generat
...
+
     if args.evaluation and validation_generator:
         if args.dataset_type == 'coco':
             from ..callbacks.coco import CocoEval
 
             # use prediction model for evaluation
             evaluation = CocoEval(validation_generator)
         else:
-            evaluation = Evaluate(validation_generator)
+            if args.tensorboard_dir:
```
Comment (hgaiser) [inline keras_retinanet/bin/train.py]: Hmm what about always initializing a `tensorboard` variable, but leave it `None` if `args.tensorboard_dir` is not set. That way this line can remain (without if):

```python
evaluation = Evaluate(validation_generator, tensorboard=tensorboard)
```

File: keras_retinanet/bin/train.py (hunk 1 above)
Comment (awilliamson) [reply keras_retinanet/bin/train.py]: Or perhaps `tb_callback` to more closely reflect what the Object is?

Comment (de-vri-es) [reply keras_retinanet/bin/train.py]: I'd prefer `tensorboard_callback` then. `tb` is a bit cryptic to me.

**Comment by mxvs** [comment]
Review remarks have been fixed, I went with the naming suggestion offered by @de-vri-es

**Comment by hgaiser** [comment]
LGTM, thanks!

**Pull request merged**